core/sr-primitives/src/lib.rs

```diff
@@ -258,6 +258,82 @@ impl From<codec::Compact<Perbill>> for Perbill {
 	}
 }
 
+/// Perquill is parts-per-quintillion. It stores a value between 0 and 1 in fixed point and
+/// provides a means to multiply some other value by that.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq)]
+pub struct Perquill(u64);
+
+impl Perquill {
+	/// Returns the internal u64 value
+	pub fn extract(&self) -> u64 { self.0 }
```

Comment:
Deref is meant to be for smart pointers

Reply:
Well... Its currently changed to `Deref` instead of `extract()`. 

Docs says: 
> Implementing Deref for smart pointers makes accessing the data behind them convenient, ... 

From which I understand it is _good_ for smart pointers, not exclusive. 

Though I am personally in favor of `.extract()` or anything else since I hate this type of code. 
https://github.com/paritytech/substrate/pull/1915/commits/747d6500d9c8332bd4d63e3c34542f82fed89b33#diff-ba0a45203e7c14aedb758fe4d3653b35R138